src/Sulu/Bundle/ContentBundle/Search/EventListener/ReindexListener.php

```diff
@@ -106,39 +89,34 @@ public function onIndexRebuild(IndexRebuildEvent $event)
             $this->purgeContentIndexes($output);
         }
 
-        $rows = $result->getRows();
-
+        $documents = $query->execute();
         $progress = new ProgressHelper();
-        $progress->start($output, count($rows));
+        $progress->start($output, count($documents));
 
-        /** @var Row $row */
-        foreach ($rows as $row) {
-            $node = $row->getNode('a');
-
-            $locales = $this->nodeHelper->getLanguagesForNode($node);
+        foreach ($documents as $document) {
+            $locales = $this->inspector->getLocales($document);
 
             foreach ($locales as $locale) {
                 try {
-                    $structure = $this->contentMapper->loadByNode($node, $locale, null, false, true, false);
-                    $structureClass = get_class($structure);
-
-                    if (!isset($count[$structureClass])) {
-                        $count[$structureClass] = array(
-                            'indexed' => 0,
-                        );
-                    }
+                    $this->documentManager->find($document->getUuid(), $locale);
```

Comment:
This will simply switch the document to the locale as the PHPCR node is already in the registry.